### DISCUSSION TITLE
Adding Patents

### DISCUSSION THREAD

**Discussion by BeefcakeTheMighty** [papis/papis#725] [Q&A]
Can patents be added as well? I've tried several different patents trying to automatically pull their metadata using patent numbers, but haven't had any luck.

**Comment by alexfikl** [comment]
To my knowledge, papis doesn't have support for anything like that at the moment. It shouldn't be too hard to add an importer for patent metadata if you know of a nice source with an open API.

**Reply by BeefcakeTheMighty** [reply]
The only one I'm familiar with is Google's. 

I might take a stab at building it out myself. If I make an real progress, I'll submit a pull request for consideration.

**Reply by alexfikl** [reply]
That sounds great! Feel free to ask any questions if you get stuck somewhere.

From a quick look at a random patent page on Google Patents
https://patents.google.com/patent/US11759238B2
it seems like they have a lot of `meta` tags in the [Dublin Core](https://en.wikipedia.org/wiki/Dublin_Core) format, so you may have some luck just downloading the relevant page and scraping it using 
https://github.com/papis/papis/blob/04f6b62bca4c84b4ad4d3eb97fc0d7d4742a3ba5/papis/downloaders/base.py#L101

There's a few examples in the `papis/downloaders` folder that use that already, if you need inspiration!

**Reply by BeefcakeTheMighty** [reply]
Awesome!! Thanks you!
